src/server/journal/pending_buf.h

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <absl/container/inlined_vector.h>
+
+#include <algorithm>
+#include <deque>
+
+namespace dfly {
+
+class PendingBuf {
+ public:
+  struct Buf {
+    size_t mem_size = 0;
+    absl::InlinedVector<std::string, 8> buf;
+
+#ifdef UIO_MAXIOV
+    static constexpr size_t max_buf_size = UIO_MAXIOV;
+#elif
+    static constexpr size_t max_buf_size = 1024;
+#endif
+  };
+
+  PendingBuf() : bufs_(1) {
+  }
+
+  bool empty() const {
+    return std::all_of(bufs_.begin(), bufs_.end(), [](const auto& b) { return b.buf.empty(); });
+  }
+
+  void push(std::string str) {
```

Comment:
```suggestion
  void Push(std::string str) {
```
I get lowercase `empty()` and `size()` to be stl-like, but let's use upper case for not clearly STL container-like methods